ca/config.go

```diff
@@ -157,6 +157,11 @@ func (s *SecurityConfig) UpdateRootCA(rootCA *RootCA, externalCARootPool *x509.C
 
 	s.rootCA = rootCA
 	s.externalCAClientRootPool = externalCARootPool
+
+	s.externalCA.mu.Lock()
+	s.externalCA.rootCA = rootCA
+	s.externalCA.mu.Unlock()
```

Comment:
The `rootCA` field is used without holding the mutex in `Sign`.

Also, could you please add a setter method for this? There is already one for `URLs`.